SPECIFICATION.md

```diff
@@ -0,0 +1,93 @@
+<h1>Provider Specification</h1>
+
+[TOC]
+
+# General
+- **name** Any record name supplied to the provider must be the full hostname.
```

Comment:
I'm not sure this line makes sense. Different providers have different requirements, some need FQDN, some support full name and relative names. We can't control the API's interface. 

On the client input/CLI side though, I've been focusing on being as flexible as possible. Given a FDQN, full name or relative name, lexicon should do the right thing. It's why I have tests like these: https://github.com/AnalogJ/lexicon/blob/master/tests/providers/integration_tests.py#L75-L91

Reply:
Requiring that the client provides FQDN and pushing the job of handling the provider requirements to the provider will make for a more consistent API. The user will not have to know whether to provide FQDN or relative based on different providers.